state/settings_test.go

```diff
@@ -594,3 +568,38 @@ func (s *SettingsSuite) TestUpdatingInterfaceSliceValue(c *gc.C) {
 	c.Assert(found, gc.Equals, true)
 	c.Assert(value, gc.DeepEquals, []interface{}{"bar1", "bar2"})
 }
+
+func (s *SettingsSuite) TestApplyAndRetrieveChanges(c *gc.C) {
+	s1, err := s.createSettings(s.key, map[string]interface{}{
+		"foo":    "bar",
+		"alpha":  "beta",
+		"number": 1,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	_, err = s1.Write()
+	c.Assert(err, jc.ErrorIsNil)
+
+	s2, err := s.readSettings()
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Add, update, update one not present, delete, delete one not present,
+	// leave one alone.
+	s2.applyChanges(coresettings.ItemChanges{
+		coresettings.MakeModification("foo", "no-matter", "new-bar"),
+		coresettings.MakeModification("make", "no-matter", "new"),
+		coresettings.MakeDeletion("alpha", "no-matter"),
+		coresettings.MakeDeletion("what", "the"),
+		coresettings.MakeAddition("new", "noob"),
+	})
+
+	// Updating one not present = addition, deleting one not present = no-op.
+	exp := coresettings.ItemChanges{
```

Comment:
Where did "number" go?

Reply:
Number is not affected by the application of the change-set, so it is not present in a delta; I.e. it remains as is.